### DISCUSSION TITLE
How to you install binaries on new machine?

### DISCUSSION THREAD

**Discussion by joeitu** [twpayne/chezmoi#1751] [Q&A]
As chezmoi is a great solution to share text file across multiple machines, a side problem is how do you port your favorite software to your new machine?

**Comment by nandalopes** [comment]
You can use this tips combined:

- [Install packages with scripts](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#install-packages-with-scripts)
- [Run a script when the contents of another file changes](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#run-a-script-when-the-contents-of-another-file-changes)

I'd suggest you split your scripts by package manager and by action (configure sources and install packages, for example).
[Here](https://github.com/nandalopes/dotfiles/tree/friends/.chezmoiscripts) is an example from my public dotfiles.

**Comment by twpayne** [comment]
#1733 discusses the same question.

**Comment by felipecrs** [comment]
I do this with `.chezmoiexternal.yaml`. If you check mine, you will notice that I use it to install stuff like:

[deno, volta, docker-compose, docker-buildx](https://github.com/felipecrs/dotfiles/blob/2672b1a9e9d76047c0f341b7c4af7ffeb7c69f06/home/.chezmoiexternal.yaml#L75-L94) and [some more things](https://github.com/felipecrs/dotfiles/blob/2672b1a9e9d76047c0f341b7c4af7ffeb7c69f06/root/.chezmoiexternal.yaml#L52-L82).
